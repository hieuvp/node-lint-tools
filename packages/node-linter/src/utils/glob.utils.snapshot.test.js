const glob = require('glob');

const { getPatternByExtension } = require('./glob.utils');

describe('getPatternByExtension', () => {
  const extensions = [
    ['js'],
    ['json'],
    ['yml'],
    ['md'],
    ['js', 'yml'],
    ['json', 'md'],
    ['js', 'json', 'yml'],
    ['js', 'json', 'yml', 'md']
  ];

  extensions.forEach(args => {
    const pattern = getPatternByExtension(args);

    it(`should match snapshot when calling with ${JSON.stringify(args)}`, () => {
      expect(pattern).toMatchSnapshot();
    });

    it(`should list all files with extension ${JSON.stringify(args)}`, () => {
      const files = glob.sync(pattern);
      expect(files).toMatchSnapshot();
    });
  });
});
