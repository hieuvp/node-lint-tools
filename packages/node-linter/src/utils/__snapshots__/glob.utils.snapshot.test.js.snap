// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPatternByExtension should list all files with extension ["js","json","yml","md"] 1`] = `
Array [
  ".eslintrc.js",
  ".travis.yml",
  "jest.config.js",
  "lerna.json",
  "package-lock.json",
  "package.json",
  "README.md",
  "runfile.js",
]
`;

exports[`getPatternByExtension should list all files with extension ["js","json","yml"] 1`] = `
Array [
  ".eslintrc.js",
  ".travis.yml",
  "jest.config.js",
  "lerna.json",
  "package-lock.json",
  "package.json",
  "runfile.js",
]
`;

exports[`getPatternByExtension should list all files with extension ["js","yml"] 1`] = `
Array [
  ".eslintrc.js",
  ".travis.yml",
  "jest.config.js",
  "runfile.js",
]
`;

exports[`getPatternByExtension should list all files with extension ["js"] 1`] = `
Array [
  ".eslintrc.js",
  "jest.config.js",
  "runfile.js",
]
`;

exports[`getPatternByExtension should list all files with extension ["json","md"] 1`] = `
Array [
  "lerna.json",
  "package-lock.json",
  "package.json",
  "README.md",
]
`;

exports[`getPatternByExtension should list all files with extension ["json"] 1`] = `
Array [
  "lerna.json",
  "package-lock.json",
  "package.json",
]
`;

exports[`getPatternByExtension should list all files with extension ["md"] 1`] = `
Array [
  "README.md",
]
`;

exports[`getPatternByExtension should list all files with extension ["yml"] 1`] = `
Array [
  ".travis.yml",
]
`;

exports[`getPatternByExtension should match snapshot when calling with ["js","json","yml","md"] 1`] = `"{*.js,.*.js,*.json,.*.json,*.yml,.*.yml,*.md,.*.md}"`;

exports[`getPatternByExtension should match snapshot when calling with ["js","json","yml"] 1`] = `"{*.js,.*.js,*.json,.*.json,*.yml,.*.yml}"`;

exports[`getPatternByExtension should match snapshot when calling with ["js","yml"] 1`] = `"{*.js,.*.js,*.yml,.*.yml}"`;

exports[`getPatternByExtension should match snapshot when calling with ["js"] 1`] = `"{*.js,.*.js}"`;

exports[`getPatternByExtension should match snapshot when calling with ["json","md"] 1`] = `"{*.json,.*.json,*.md,.*.md}"`;

exports[`getPatternByExtension should match snapshot when calling with ["json"] 1`] = `"{*.json,.*.json}"`;

exports[`getPatternByExtension should match snapshot when calling with ["md"] 1`] = `"{*.md,.*.md}"`;

exports[`getPatternByExtension should match snapshot when calling with ["yml"] 1`] = `"{*.yml,.*.yml}"`;
